runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2873,7 +2873,13 @@ J9::Z::TreeEvaluator::generateHelperCallForVMNewEvaluators(TR::Node *node, TR::C
 TR::Register *
 J9::Z::TreeEvaluator::newObjectEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
-   if (cg->comp()->suppressAllocationInlining())
+   // If the helper symbol set on the node is TR_jitNewValue, we are (expecting to be)
+   // dealing with a value type. Since we do not fully support value types yet, always
+   // call the JIT helper to do the allocation.
+   //
+   auto* comp = cg->comp();
+   if (cg->comp()->suppressAllocationInlining() ||
+       (TR::Compiler->om.areValueTypesEnabled() && node->getSymbolReference() == comp->getSymRefTab()->findOrCreateNewValueSymbolRef(comp->getMethodSymbol())))
       return generateHelperCallForVMNewEvaluators(node, cg);
    else
       return TR::TreeEvaluator::VMnewEvaluator(node, cg);
```

Comment:
This seems a bit odd to me. Why are we re-using the `newobject` evaluator to implement `newvalue`? Shouldn't we use a separate evaluator for this? i.e.

```
TR::Register *
J9::Z::TreeEvaluator::newValueEvaluator(TR::Node * node, TR::CodeGenerator * cg)
   {
   return generateHelperCallForVMNewEvaluators(node, cg);
   }
```